From 70df9cb2b93ed3d4b7d6e201a58951c9981bbc93 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?R=C3=A9mi=20Cardona?= <remi@gentoo.org>
Date: Thu, 7 May 2009 14:19:22 +0200
Subject: [PATCH 35/78] cherry-pick the rest of "glcore: prepare for dynamic loading"

When f2c30a57e3c6169fbecdbb60ac24ad2a5e9aa8a4 was cherry-picked into the
1.5 branch, some parts were lost. This commit brings the missing bits.

Conflicts:

	configure.ac
---
 configure.ac                |    2 +-
 hw/kdrive/ephyr/Makefile.am |    4 ++++
 hw/kdrive/fake/Makefile.am  |    4 ++++
 hw/kdrive/fbdev/Makefile.am |    4 ++++
 hw/kdrive/sdl/Makefile.am   |    4 ++++
 hw/kdrive/vesa/Makefile.am  |    4 ++++
 6 files changed, 21 insertions(+), 1 deletions(-)

Index: xorg-server-1.5.3/configure.ac
===================================================================
--- xorg-server-1.5.3.orig/configure.ac	2009-09-14 19:48:11.043879237 -0700
+++ xorg-server-1.5.3/configure.ac	2009-09-14 19:49:34.903881120 -0700
@@ -1964,7 +1964,7 @@
     KDRIVE_LOCAL_LIBS="$TSLIB_LIBS $DIX_LIB $KDRIVE_LIB $KDRIVE_STUB_LIB $CONFIG_LIB"
     KDRIVE_LOCAL_LIBS="$KDRIVE_LOCAL_LIBS $FB_LIB $MI_LIB $KDRIVE_PURE_LIBS"
     KDRIVE_LOCAL_LIBS="$KDRIVE_LOCAL_LIBS $KDRIVE_OS_LIB $OS_LIB"
-    KDRIVE_LIBS="$KDRIVE_LOCAL_LIBS $XSERVER_SYS_LIBS"
+    KDRIVE_LIBS="$KDRIVE_LOCAL_LIBS $XSERVER_SYS_LIBS $GLX_SYS_LIBS"
 
     # check if we can build Xephyr
     PKG_CHECK_MODULES(XEPHYR, $XEPHYR_REQUIRED_LIBS, [xephyr="yes"], [xephyr="no"])
Index: xorg-server-1.5.3/hw/kdrive/ephyr/Makefile.am
===================================================================
--- xorg-server-1.5.3.orig/hw/kdrive/ephyr/Makefile.am	2008-11-05 08:52:17.000000000 -0800
+++ xorg-server-1.5.3/hw/kdrive/ephyr/Makefile.am	2009-09-14 19:49:34.903881120 -0700
@@ -67,6 +67,10 @@
 Xephyr_SOURCES = \
 	ephyrinit.c
 
+if GLX
+Xephyr_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+endif
+
 Xephyr_LDADD = 						\
 	libxephyr.a					\
 	libxephyr-hostx.a			        \
Index: xorg-server-1.5.3/hw/kdrive/fake/Makefile.am
===================================================================
--- xorg-server-1.5.3.orig/hw/kdrive/fake/Makefile.am	2008-07-11 08:12:13.000000000 -0700
+++ xorg-server-1.5.3/hw/kdrive/fake/Makefile.am	2009-09-14 19:49:34.903881120 -0700
@@ -16,6 +16,10 @@
 Xfake_SOURCES = \
 	fakeinit.c
 
+if GLX
+Xfake_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+endif
+
 Xfake_LDADD = 						\
 	libfake.a					\
 	@KDRIVE_LIBS@					\
Index: xorg-server-1.5.3/hw/kdrive/fbdev/Makefile.am
===================================================================
--- xorg-server-1.5.3.orig/hw/kdrive/fbdev/Makefile.am	2008-07-11 08:12:13.000000000 -0700
+++ xorg-server-1.5.3/hw/kdrive/fbdev/Makefile.am	2009-09-14 19:49:34.903881120 -0700
@@ -14,6 +14,10 @@
 Xfbdev_SOURCES = \
 	fbinit.c
 
+if GLX
+Xfbdev_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+endif
+
 Xfbdev_LDADD = 						\
 	libfbdev.a					\
 	@KDRIVE_LIBS@
Index: xorg-server-1.5.3/hw/kdrive/sdl/Makefile.am
===================================================================
--- xorg-server-1.5.3.orig/hw/kdrive/sdl/Makefile.am	2008-07-11 08:12:13.000000000 -0700
+++ xorg-server-1.5.3/hw/kdrive/sdl/Makefile.am	2009-09-14 19:49:34.903881120 -0700
@@ -7,6 +7,10 @@
 
 Xsdl_SOURCES = sdl.c
 
+if GLX
+Xsdl_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+endif
+
 Xsdl_LDADD = \
 	@KDRIVE_LIBS@                                  \
 	@XSDL_LIBS@
Index: xorg-server-1.5.3/hw/kdrive/vesa/Makefile.am
===================================================================
--- xorg-server-1.5.3.orig/hw/kdrive/vesa/Makefile.am	2008-11-05 08:52:17.000000000 -0800
+++ xorg-server-1.5.3/hw/kdrive/vesa/Makefile.am	2009-09-14 19:49:34.903881120 -0700
@@ -19,6 +19,10 @@
 Xvesa_SOURCES = \
 	vesainit.c
 
+if GLX
+Xvesa_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+endif
+
 Xvesa_LDADD = \
 	libvesa.a 				\
 	@KDRIVE_LIBS@
